### PR TITLE
CU-864dhrxv7 - Implement Sentry or similar tool to track errors in pr…

### PR DESCRIPTION
**Pull request by LeonardoDizConde**

### CODE REVIEW

**Comment by lllwvlvwlll** [comment]
Task linked: [CU-864dhrxv7 NEON3 - Implement Sentry or similar tool to track errors in production](https://app.clickup.com/t/864dhrxv7)

File: src/renderer/src/setup/sentryReact.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const isProductionMode = (): boolean => {
+  const env = window['env']
+  return env && env['SENTRY_DNS'] && env['ENVIRONMENT'] && env['ENVIRONMENT'] == EnvironmentMode.PROD
+}
+export function setupSentryReact() {
+  if (isProductionMode()) {
+    ElectronRedererSentry.init(
+      {
+        dsn: window['env']['SENTRY_DNS'],
```
Comment (raulduartep) [inline src/renderer/src/setup/sentryReact.ts]:
```suggestion
        dsn: window.env.SENTRY_DNS
```

You will need to type in the src/renderer/src/@types/window.ts file

File: src/renderer/src/setup/sentryReact.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { createBrowserRouter } from 'react-router-dom'
+import * as ElectronRedererSentry from '@sentry/electron/renderer'
+import * as ReactSentry from '@sentry/react'
+
+import { EnvironmentMode } from '../../../common/EnvironmentMode'
+import { sentryCommonConfig } from '../../../common/GlobalConfig'
+
+const isProductionMode = (): boolean => {
+  const env = window['env']
+  return env && env['SENTRY_DNS'] && env['ENVIRONMENT'] && env['ENVIRONMENT'] == EnvironmentMode.PROD
```
Comment (raulduartep) [inline src/renderer/src/setup/sentryReact.ts]: Same thing as the comment below

File: src/common/GlobalConfig.ts
Comment (raulduartep) [inline src/common/GlobalConfig.ts:1]: Rename this file to sentryCommonConfig

File: src/common/EnvironmentMode.ts
Comment (raulduartep) [inline src/common/EnvironmentMode.ts:1]: I think it would be better if this enum didn't exist, try typing the env instead

**Pull request merged**